pkgs/development/haskell-modules/configuration-ghc-9.4.x.nix

```diff
@@ -63,13 +63,20 @@ in {
 
   hashable-time = doJailbreak super.hashable-time;
   libmpd = doJailbreak super.libmpd;
+  lens-family-th = doJailbreak super.lens-family-th;  # template-haskell <2.19
+
   # generically needs base-orphans for 9.4 only
   base-orphans = dontCheck (doDistribute super.base-orphans);
 
   # the dontHaddock is due to a GHC panic. might be this bug, not sure.
   # https://gitlab.haskell.org/ghc/ghc/-/issues/21619
   hedgehog = dontHaddock super.hedgehog;
 
+  # Cherry-pick GHC 9.4 changes from hnix master branch
+  hnix = appendPatches [
+    ./patches/hnix-compat-for-ghc-9.4.patch
```

Comment:
Could you fetch this patch from upstream somehow?  Maybe pulling a diff for a single file from a specific commit?  There should be a few examples of this in the other `configuration-*` files.

We generally try to pull patches from upstream if at all possible rather than committing them to Nixpkgs.

Other than that, this PR LGTM!  Thanks

Reply:
Unfortunately I couldn't fetch it from upstream. There was no PR, and the relevant commit from master branch had other changes which didn't apply cleanly to the 0.16 sources. Taking the latest master branch revision also wasn't an option because of API changes.

Reply:
Ah, I see, that's unfortunate.

Could you write that in a comment in the code and link to the commits that you had manually pulled these changes from?

Reply:
I added a message to the patch noting that it's a backport.

Reply:
Thanks, I appreciate it

Reply:
@rvl `excludes`, `includes` usually helps as long as the change itself applies cleanly.